src/main/java/org/hawkular/client/android/auth/ModifiedAuthzModule.java

```diff
@@ -0,0 +1,197 @@
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.hawkular.client.android.auth;
+
+import java.net.URI;
+import java.net.URL;
+import java.util.UUID;
+
+import org.hawkular.client.android.activity.LoginActivity;
+import org.jboss.aerogear.android.authorization.AuthzModule;
+import org.jboss.aerogear.android.authorization.oauth2.OAuth2AuthzSession;
+import org.jboss.aerogear.android.core.Callback;
+import org.jboss.aerogear.android.pipe.http.HeaderAndBody;
+import org.jboss.aerogear.android.pipe.http.HttpException;
+import org.jboss.aerogear.android.pipe.http.HttpProvider;
+import org.jboss.aerogear.android.pipe.http.HttpRestProvider;
+import org.jboss.aerogear.android.pipe.module.AuthorizationFields;
+import org.jboss.aerogear.android.pipe.module.ModuleFields;
+import org.jboss.aerogear.android.store.DataManager;
+import org.jboss.aerogear.android.store.generator.IdGenerator;
+import org.jboss.aerogear.android.store.sql.SQLStore;
+import org.jboss.aerogear.android.store.sql.SQLStoreConfiguration;
+
+import android.app.Activity;
+import android.content.Context;
+import android.os.AsyncTask;
+import android.util.Base64;
+import android.util.Log;
+
+/**
+ * ModifiedAuthzModule.
+ * <p/>
+ * Performs all related work of generating new key and secret pair, Storing them as account,
+ * Retrieving stored account, Deleting stored account.
+ */
+
+public class ModifiedAuthzModule implements AuthzModule {
+
+    Context applicationContext;
+    private SQLStore<OAuth2AuthzSession> sessionStore;
+
+    public ModifiedAuthzModule(Context context) {
+        applicationContext = context;
+        sessionStore = openSessionStore();
+    }
+
+    @Override
+    public boolean isAuthorized() {
+        return true;
+    }
+
+    @Override
+    public boolean hasCredentials() {
+        OAuth2AuthzSession storedAccount = sessionStore.read("hawkular");
+        if (storedAccount != null) {
+            return true;
+        }
+        return false;
+    }
+
+    @Override
+    public void requestAccess(final Activity activity, final Callback<String> callback) {
+
+        Exception exception;
+        String json;
+        if (hasCredentials()) {
+            OAuth2AuthzSession storedAccount = sessionStore.read("hawkular");
+            final String token = storedAccount.getAccessToken();
+            try {
+                callback.onSuccess(token);
+            } catch (Exception e) {
+                callback.onFailure(e);
+            }
+        } else {
+
+            AsyncTask<Void, Void, HeaderAndBody> task = new AsyncTask<Void, Void, HeaderAndBody>() {
```

Comment:
Most likely you can get away with a simple thread or single threaded executor. This way you can avoid storing exceptions in fields.